src/Symfony/Component/ErrorHandler/DebugClassLoader.php

```diff
@@ -849,6 +849,26 @@ private function setReturnType(string $types, string $class, string $method, str
         $docTypes = [];
 
         foreach ($typesMap as $n => $t) {
+            if (str_contains($n, '::')) {
+                [$definingClass, $constantName] = explode('::', $n, 2);
+                $definingClass = match ($definingClass) {
+                    'self', 'static', 'parent' => $class,
+                    default => $definingClass,
+                };
+
+                if (!\defined($definingClass.'::'.$definingClass)) {
+                    return;
+                }
+
+                $constant = new \ReflectionClassConstant($definingClass, $constantName);
+
+                if (method_exists(\ReflectionClassConstant::class, 'getType') && $constantType = $constant->getType()) {
+                    $n = $constantType->getName();
```

Comment:
is the constant type guaranteed to be a ReflectionNamedType in PHP ? https://github.com/php/php-src/blob/9345582471d5dfa8b783b1e666d8b76620f542f9/ext/reflection/php_reflection.stub.php#L626C32-L626C47 seems to answer "no"

Reply:
I guess the stub is not correct here. I added a test covering this line and `$constantType` is indeed an instance of `ReflectionNamedType`.

Reply:
well, if the type you use is a named type, it will. But do constant types allows using union or intersection types ? Those are not named types.

Reply:
Yes, they can be union or intersection types (see my edit above). I've opted to ignore them for now in this PR. We can consider to add support for them in the future (similar to the support for wildcards).